flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -145,20 +148,20 @@ public void close() throws IOException {
 	 * checkpoint and enqueues the corresponding reader for this consumer (if not enqueued yet).
 	 *
 	 * @param receiverId The input channel id to identify the consumer.
-	 * @param operation The operation to be performed (add credit or resume data consumption).
+	 * @param function The operation to be performed (add credit or resume data consumption).
 	 */
 	void addCreditOrResumeConsumption(
 			InputChannelID receiverId,
-			Consumer<NetworkSequenceViewReader> operation) throws Exception {
+			Function<NetworkSequenceViewReader, Boolean> function) throws Exception {
 		if (fatalError) {
 			return;
 		}
 
 		NetworkSequenceViewReader reader = allReaders.get(receiverId);
 		if (reader != null) {
-			operation.accept(reader);
+			boolean needAnnounceBacklog = function.apply(reader);
```

Comment:
Actually the logic for adding credit does not need `needAnnounceBacklog`, so it might bring trouble to understand the logic of addCredit by reusing the common codes here.

One possible solution is to call `enqueueAvailableReader` in reader stack while applying the function, then the `NetworkSequenceViewReader#addCredit` and `NetworkSequenceViewReader#resumeConsumption` can judge the separate conditions before calling `enqueueAvailableReader`.

Reply:
To do so we can also avoid adjusting to return unnecessary boolean value for `NetworkSequenceViewReader#addCredit`